packages/navigation/src/hooks/useWebviewNavigate.ts

```diff
@@ -71,15 +93,51 @@ function getAction(
     return StackActions.replace(action.payload.name, action.payload.params);
   } else {
     const { name, params } = action.payload;
-    const key = getKeyFromParams(params);
-    return CommonActions.navigate({
-      name,
-      key,
-      params,
-    });
+    if (action.type === 'NAVIGATE') {
+      // const key = getKeyFromParams(params);
+      return CommonActions.navigate({
+        name,
+        // key,
+        params,
+      });
+    }
+    return StackActions.push(name, params);
   }
 }
 
+function getMinimalAction(
+  action: NavigationAction,
+  state: NavigationState
+): NavigationAction {
+  let currentAction = action;
+  let currentState:
+    | NavigationState
+    | PartialState<NavigationState>
+    | undefined = state;
+
+  while (
+    currentAction.payload &&
+    'name' in currentAction.payload &&
+    currentState?.routes[currentState.index ?? -1]?.name ===
+      currentAction.payload.name
```

Comment:
Should you call `navigate` when you change the params? It sounds like a `push` then from `@react-navigation` point of view, but I may be wrong.

Reply:
I was thinking in case of a nested navigator where the parent has a param glob in the path definitions, eg: `/users/:slug/images/:id` where they are built with nested navigators such as:

```
routes: {
   Users: {
     path: "users/:slug",
     screens: {
       Images: "images/:id",
     }
   }
}
```

If navigating from eg. `/users/bob/images/123` to `/users/alice/images/456`, I suppose we'd like to **_not_** remove the `Users` screen as part of the minimal action as the params differ but instead dispatch the whole navigation action and so that a new `Users` screen is pushed onto the stack.

Reply:
Yeah, it makes sense for sure. and it should push new `Users` navigator with new `Images` screen then yeah?

Reply:
Yeah, exactly.